packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,66 @@
 import Objection from 'objection';
+import {SignedState, calculateChannelId, Objective} from '@statechannels/wallet-core';
 
 import {SignState} from '../protocols/actions';
-import {Channel, SyncState} from '../models/channel';
+import {Channel, SyncState, RequiredColumns} from '../models/channel';
+import {SigningWallet} from '../models/signing-wallet';
+import {logger} from '../logger';
+import {addHash} from '../state-utils';
+import knex from '../db/connection';
 
 export const Store = {
+  transaction: async function(callback: () => Promise<void>): Promise<void> {
+    return knex.transaction(async () => {
+      await callback();
+    });
+  },
```

Comment:
```ts
transaction: (callback: () => Promise<void>): Promise<void> => knex.transaction(callback),
```
Maybe more succinct? Changes the return type I think. Why do we want this; is it necessary to lock the _entire_ store for any use case?

Reply:
FYI: SQL transactions do not lock the entire store. Postgresql will acquire locks as necessary during the transaction, such as row-level write locks if on multiple rows, if updating multiple channels within a single transaction.

This is different from indexedDB transactions, which are scoped to (named) stores.
- write transactions acquire a read/write lock on the entirety of all stores that it's scoped to
- read transactions acquire a read lock, which do not prevent others from reading

Reply:
Neat. Is there any reason to use `Channel.transaction` ever if we can just use this general purpose `Store.transaction`? 

Also, it seems like `Store.transaction` is not "backend-agnostic" then, since it doesn't make sense in the context of an IndexedDB backend _(or, maybe it does, if you implement some logic to check which tables should be modified in the IndexedDB case?)_. Maybe we should just be careful about what methods we tag on to the `Store` since a goal is to figure out how much of this can be re-used in the browser wallet.

Reply:
`Channel.transaction` is just a convenience method for starting a database transaction. It's not special to the `Channel` model in any way.

It's not really possible to have a db-agnostic transaction, since database management systems all implement transactions differently. The store needs to be up front about what properties of transactions it is using. The important property is the ability to roll back.

The browser wallet also assumed that transactions could be scoped to "object stores", and that write locks would Lock the entire object store. That is something you can implement in Postgres, but it is not the default behaviour (for good reason!)